packages/flutter_tools/test/web.shard/hot_reload_web_errors_test.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@Tags(<String>['flutter-test-driver'])
+library;
+
+import 'package:file/file.dart';
+
+import '../integration.shard/test_data/hot_reload_const_project.dart';
+import '../integration.shard/test_driver.dart';
+import '../integration.shard/test_utils.dart';
+import '../src/common.dart';
+
+void main() {
+  late Directory tempDir;
+  final HotReloadConstProject project = HotReloadConstProject();
+  late FlutterRunTestDriver flutter;
+
+  final List<String> additionalCommandArgs = <String>[
+    '--extra-front-end-options=--dartdevc-canary,--dartdevc-module-format=ddc',
+  ];
+
+  setUp(() async {
+    tempDir = createResolvedTempDirectorySync('hot_reload_test.');
+    await project.setUpIn(tempDir);
+    flutter = FlutterRunTestDriver(tempDir);
+  });
+
+  tearDown(() async {
+    await flutter.stop();
+    tryToDelete(tempDir);
+  });
+
+  testWithoutContext(
+    'hot reload displays an error message when removing a field from a const class',
```

Comment:
The expected error string is actually a different message about removing all const constructors making a const class no longer const.